### PR TITLE
Bug 1615330 - Support filtering by test paths rather than manifest paths

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

**Comment by armenzg** [comment]
@camd you can see test being filtered in [here](https://treeherder-test-filteri-vv5oe4.herokuapp.com/#/jobs?repo=try&test_paths=devtools%2Fclient%2Fframework%2Fbrowser-toolbox%2Ftest%2Fbrowser_browser_toolbox.js&revision=de74a9cb8389c0f88ae8a716887ce3d91c8c9031)

**Comment by armenzg** [comment]
@camd using a test path to filter: https://treeherder-test-filteri-vv5oe4.herokuapp.com/#/jobs?repo=try&revision=8c9e265b85cba570943e115184d68a20a3ed2627&test_paths=browser/components/payments/test/mochitest/formautofill/test_editCreditCard.html

**Comment by camd** [comment]
This looks like it needs a rebase, afaict.  Though I'm having a hard time checking this out locally.  Not quite sure why.  Seems like there's a branch name conflict, which I didn't think was possible.  :)

File: Procfile [hunk 1]
```diff
@@ -36,10 +36,9 @@ celery_scheduler: REMAP_SIGTERM=SIGQUIT newrelic-admin run-program celery beat -
 # to the `store_pulse_{pushes,jobs}` queues for `worker_store_pulse_data` to process.
 # NB: These should not be scaled up to more than 1 of each.
 # TODO: Merge these two listeners into one since they use so little CPU each (bug 1530965).
-pulse_listener_pushes: newrelic-admin run-program ./manage.py pulse_listener_pushes
-pulse_listener_tasks: newrelic-admin run-program ./manage.py pulse_listener_tasks
+pulse_listener_tasks_pushes: newrelic-admin run-program ./manage.py pulse_listener
```
Comment (camd) [inline Procfile]: This change to merge pulse tasks/pushes seems orthogonal to the filtering by test paths.  Did these commits get mixed in by mistake?

File: tests/ui/job-view/Push_test.jsx [hunk 2]
```diff
@@ -49,21 +75,22 @@ describe('Push', () => {
...
-          'devtools/client/inspector/grids/test/browser.ini',
-          'devtools/client/inspector/rules/test/browser.ini',
-          'devtools/client/jsonview/test/browser.ini',
-        ],
-      },
-    );
+    const tcUrl =
+      'https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.v2.autoland.revision.d5b037941b0ebabcc9b843f24d926e9d65961087.taskgraph.decision/artifacts/public';
+    // XXX: Fix this to re-enable test
+    // I need to figure out the right options to get a gzip blob
```
Comment (armenzg) [inline tests/ui/job-view/Push_test.jsx]: The changes here don't actually do anything since the test is disable, however, I wanted *some* of my investigation not be lost so Ieft it here.

Comment (camd) [reply tests/ui/job-view/Push_test.jsx]: Sounds good.  :)  I know just what you mean.
